spanner/spanner_snippets/spanner/integration_test.go

```diff
@@ -90,6 +91,24 @@ func initBackupTest(t *testing.T, id, dbName string) (restoreDBName, backupID, c
 		t.Logf("database %s exists in state %s. delete result: %v", db.GetName(), db.GetState().String(),
 			adminClient.DropDatabase(ctx, &adminpb.DropDatabaseRequest{Database: restoreDBName}))
 	}
+	client, err := spanner.NewClient(ctx, restoreDatabaseID)
```

Comment:
I'm not sure if I'm missing something here, or if this just coincidentally goes well (sometimes): We are creating a client to read from `restoreDatabaseID` here. On the lines above we check whether that database exists, and if it does, we drop it:
```go
	if db, err := adminClient.GetDatabase(ctx, &adminpb.GetDatabaseRequest{Name: restoreDBName}); err == nil {
		t.Logf("database %s exists in state %s. delete result: %v", db.GetName(), db.GetState().String(),
			adminClient.DropDatabase(ctx, &adminpb.DropDatabaseRequest{Database: restoreDBName}))
	}
```

I would therefore expect this client to return an error that the database does not exist.

Reply:
I think you are correct, it is intriguing that the integration tests are passing.